core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -647,16 +590,31 @@ private object RTS {
     private final def addFailures(ts: List[Throwable]): Unit = {
       val oldStatus = status.get
       oldStatus match {
-        case x @ Executing(_, ts0, _) =>
-          if (!status.compareAndSet(oldStatus, x.copy(defects = ts0 ++ ts))) addFailures(ts) else ()
+        case x @ Executing(cause, _) =>
```

Comment:
`accumFailures`, `addFailures`, and `addDefects` can all be deleted. Any place using them needs to be changed (I think I caught all of them in the above review).

Reply:
This is bloody great.